httpx/_config.py

```diff
@@ -42,150 +40,102 @@ class UnsetType:
 UNSET = UnsetType()
 
 
-def create_ssl_context(
-    cert: typing.Optional[CertTypes] = None,
-    verify: VerifyTypes = True,
-    trust_env: bool = True,
-    http2: bool = False,
-) -> ssl.SSLContext:
-    return SSLConfig(
-        cert=cert, verify=verify, trust_env=trust_env, http2=http2
-    ).ssl_context
-
-
-class SSLConfig:
-    """
-    SSL Configuration.
-    """
-
+class SSLContext(ssl.SSLContext):
     DEFAULT_CA_BUNDLE_PATH = Path(certifi.where())
 
-    def __init__(
-        self,
-        *,
-        cert: typing.Optional[CertTypes] = None,
+    def __new__(
+        cls,
+        protocol: ssl._SSLMethod = ssl.PROTOCOL_TLS_CLIENT,
         verify: VerifyTypes = True,
-        trust_env: bool = True,
-        http2: bool = False,
-    ) -> None:
-        self.cert = cert
-        self.verify = verify
-        self.trust_env = trust_env
-        self.http2 = http2
-        self.ssl_context = self.load_ssl_context()
+        cert: typing.Optional[CertTypes] = None,
+    ) -> "SSLContext":
```

Comment:
I'm curious why we're overriding `__new__` instead of `__init__` here?

Reply:
Yes, it should definitely be in __init__. I believe I encountered some issues and have temporarily moved it to __new__ to see how it looks in general.